tests/test_logger.py

```diff
@@ -0,0 +1,149 @@
+    def test_button_ok_clicked(self):
+        with mock.patch('iquip.apps.logger.ConfirmClearingFrame.buttonOKClicked') as mocked_method:
+            app = logger.LoggerApp(name="name", parent=QObject())
+            app.confirmFrame.buttonBox.accepted.emit()
+            mocked_method.assert_called_once()
+
+    def test_button_cancel_clicked(self):
+        method_dir = 'iquip.apps.logger.ConfirmClearingFrame.buttonCancelClicked'
+        with mock.patch(method_dir) as mocked_method:
+            app = logger.LoggerApp(name="name", parent=QObject())
+            app.confirmFrame.buttonBox.rejected.emit()
+            mocked_method.assert_called_once()
+
+
+class LoggerAppTest(unittest.TestCase):
+    """Unit tests for the LoggerApp class."""
+
+    def setUp(self):
+        self.qapp = QApplication([])
+
+    def tearDown(self):
+        del self.qapp
+
+    def test_set_level(self):
+        """Tests for the LoggerApp's method setLevel.
+        
+        For vaild input, tests whether level of rootLogger and handler changes succesfully.
+        For invaild input, tests whether level of rootLogger and handler remains same.
+        """
+        levels = {"DEBUG": logging.DEBUG, "INFO": logging.INFO, "WARNING": logging.WARNING,
+                  "ERROR": logging.ERROR, "CRITICAL": logging.CRITICAL}
+        app = logger.LoggerApp('name')
+        root_logger = logging.getLogger()
+        for level, value in levels.items():
+            app.setLevel(level)
+            self.assertEqual(app.handler.level, value)
+            self.assertEqual(root_logger.level, value)
+        prev_level = app.handler.level
+        non_level = "non_level"
+        app.setLevel(non_level)
+        self.assertEqual(prev_level, app.handler.level)
+        self.assertEqual(prev_level, root_logger.level)
+
+    def test_frames(self):
+        """Tests for the LoggerApp's method frames."""
+        app = logger.LoggerApp(name="name", parent=QObject())
+        self.assertEqual(app.frames(), (app.loggerFrame,))
+
+    def test_call_check_to_clear(self):
+        """Tests for the LoggerApp's method checkToClear.
+        
+        Tests that LoggerApp's method checkToclear is connected to the clearButton of LoggerFrame.
+        """
+        with mock.patch('iquip.apps.logger.LoggerApp.checkToClear') as mocked_method:
+            app = logger.LoggerApp(name="name", parent=QObject())
+            app.loggerFrame.clearButton.clicked.emit()
+            mocked_method.assert_called_once()
+
+    def test_call_clear_log(self):
+        """Tests for the LoggerApp's method clearLog.
+
+        Tests that LoggerApp's method clearLog is connected to confirmFrame's OK button.
+        """
+        with mock.patch('iquip.apps.logger.LoggerApp.clearLog') as mocked_method:
+            app = logger.LoggerApp(name="name", parent=QObject())
+            app.confirmFrame.buttonBox.accepted.emit()
+            mocked_method.assert_called_once()
+
+    def test_call_add_log(self):
```

Comment:
This is also not a unit test, because this works well only if `LoggingHandler` is implemented correctly.

Generally, tests you implemented are called functional tests and we need these, too.
Referred to `test_explorer.py`, please separate unit tests and functional tests.